src/ui/mainwindow.cc

```diff
@@ -1391,30 +1367,26 @@ void MainWindow::updateAppearances( QString const & addonStyle,
   }
 }
 
-void MainWindow::updateTrayIcon()
+void MainWindow::TrayIconUpdateOrInit()
 {
-  if ( !trayIcon && cfg.preferences.enableTrayIcon ) {
-    // Need to show it
-    trayIcon =
-      new QSystemTrayIcon( QIcon::fromTheme( "goldendict-tray", QIcon( ":/icons/programicon_old.png" ) ), this );
-    trayIcon->setContextMenu( &trayIconMenu );
-    trayIcon->show();
-
-    connect( trayIcon, &QSystemTrayIcon::activated, this, &MainWindow::trayIconActivated );
-  }
-  else if ( trayIcon && !cfg.preferences.enableTrayIcon ) {
-    // Need to hide it
-    delete trayIcon;
-
-    trayIcon = nullptr;
+  if ( !cfg.preferences.enableTrayIcon ) {
+    if ( trayIcon ) {
+      delete trayIcon;
+      trayIcon = nullptr;
+    }
   }
-  if ( trayIcon ) {
+  else {
+    if ( !trayIcon ) {
+      trayIcon = new QSystemTrayIcon( this );
+      trayIcon->setContextMenu( &trayIconMenu );
+      trayIcon->setToolTip( QApplication::applicationName() );
+      connect( trayIcon, &QSystemTrayIcon::activated, this, &MainWindow::trayIconActivated );
+    }
     // Update the icon to reflect the scanning mode
     trayIcon->setIcon( enableScanningAction->isChecked() ?
                          QIcon::fromTheme( "goldendict-scan-tray", QIcon( ":/icons/programicon_scan.png" ) ) :
                          QIcon::fromTheme( "goldendict-tray", QIcon( ":/icons/programicon_old.png" ) ) );
-
-    trayIcon->setToolTip( "GoldenDict-ng" );
+    trayIcon->show();
```

Comment:
`trayIcon->show();` is a single `bool = true`. 

Compared to the old code, the icon is only set once when the `trayIcon` previously doesn't exist.